pkg/groupsnapshot/controllers/groupsnapshot.go

```diff
@@ -477,6 +523,11 @@ func (m *GroupSnapshotController) createSnapAndDataObjects(
 
 		snap, err = k8s.Instance().CreateSnapshot(snap)
 		if err != nil {
+			// Try to replace the snapshot if it already exists
+			if errors.IsAlreadyExists(err) {
+				err = m.replaceSnapshot(snap)
```

Comment:
why delete snapshot data if we are able to replace snapshot on line 532

Reply:
Need to skip the delete below if replace succeeds. Thanks for catching that.

Reply:
@ram-infrac Please take a look again